src/arp.rs

```diff
@@ -1,59 +1,65 @@
 //! Handles parsing of Arp pakets
 
-use nom::Endianness::Big;
-use nom::{be_u8, IResult};
-
+use nom::number;
+use nom::IResult;
 use std::net::Ipv4Addr;
 
-use crate::ethernet::{to_mac_address, MacAddress};
-use crate::ipv4::to_ipv4_address;
+use crate::ethernet;
+use crate::ethernet::MacAddress;
+use crate::ipv4;
 
 #[derive(Debug, PartialEq, Eq)]
-#[cfg_attr(feature = "derive", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "derive", derive(serde::Serialize, serde::Deserialize))]
 pub enum HardwareAddressType {
     Ethernet,
     Other(u16),
 }
 
-fn to_hw_addr_type(i: u16) -> Option<HardwareAddressType> {
-    match i {
-        0x0001 => Some(HardwareAddressType::Ethernet),
-        other => Some(HardwareAddressType::Other(other)),
+impl From<u16> for HardwareAddressType {
+    fn from(raw: u16) -> Self {
+        match raw {
+            0x0001 => Self::Ethernet,
+            other => Self::Other(other),
+        }
     }
 }
 
 #[derive(Debug, PartialEq, Eq)]
-#[cfg_attr(feature = "derive", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "derive", derive(serde::Serialize, serde::Deserialize))]
 pub enum ProtocolAddressType {
     IPv4,
     Other(u16),
 }
 
-fn to_proto_addr_type(i: u16) -> Option<ProtocolAddressType> {
-    match i {
-        0x0800 => Some(ProtocolAddressType::IPv4),
-        other => Some(ProtocolAddressType::Other(other)),
+impl From<u16> for ProtocolAddressType {
+    fn from(raw: u16) -> Self {
+        match raw {
+            0x0800 => Self::IPv4,
+            other => Self::Other(other),
+        }
     }
 }
 
 #[derive(Debug, PartialEq, Eq)]
-#[cfg_attr(feature = "derive", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "derive", derive(serde::Serialize, serde::Deserialize))]
 pub enum Operation {
     Request,
     Reply,
     Other(u16),
 }
 
-fn to_operation(i: u16) -> Option<Operation> {
```

Comment:
I removed the Option, I don't see the point here. I also implement `From` trait because this conversion can't fail.